guarddog/analyzer/sourcecode/download-executable.yml

```diff
@@ -7,27 +7,45 @@ rules:
       description: Identify when a package downloads and makes executable a remote binary
     mode: taint
     pattern-sinks:
+      - pattern-either:
+          - pattern: |
+              $FILE = open(...)
+              ...
+              $FILE.write($REQUEST)
+          - pattern: open(...).write($REQUEST)
+          - pattern: |
+              with open(...) as $FILE:
+                ...
+                $FILE.write($REQUEST)
```

Comment:
instead of dropping $CHANGE_PERMISSIONS, could we add a match for `os.system(f'chmod +x $downloadedfile')` somehow in there?

Reply:
Yep ✅